include/envoy/http/header_map.h

```diff
@@ -352,6 +352,9 @@ class HeaderEntry {
   virtual const HeaderEntry* name() const PURE;                                                    \
   virtual HeaderEntry* name() PURE;                                                                \
   virtual HeaderEntry& insert##name() PURE;                                                        \
```

Comment:
do we want to remove insert in this PR? Or are there still callers we need to support?

Maybe this is worth a comment?

Reply:
Yep still need to support appending, edited the PR description.

Reply:
Should we replace this with an append() method then? Maybe you wouldn't need a non-const ref return at all then? Would that make it easier to reason about?

Reply:
I think that'd be the right approach. `insert` is probably the right name for this.